tasks/reverse_proxy.yml

```diff
@@ -1,45 +1,59 @@
 ---
+- name: remove deprecated nginx
+  docker_container:
+    name: "hub_nginx"
+    state: absent
+
+- name: remove deprecated nginx-gen
+  docker_container:
+    name: "hub_nginx_gen"
+    state: absent
+
 - name: Create a volume to hold the certificates for TLS / SSL
   docker_volume:
     name: "{{ volume_names.tls_certificates }}"
 
-- name: Start Reverse Proxy (nginx)
+- name: Start Reverse Proxy (traefik)
   docker_container:
     name: "{{ container_names.reverse_proxy }}"
     image: "{{ image_versions.reverse_proxy }}"
     restart_policy: always
     published_ports:
       - "80:80"
       - "443:443"
-    volumes_from:
-      - "{{ container_names.main }}"
+    command: |
+      --api.insecure=false
+      --api.dashboard=false
+      --api.debug=false
+      --log.level=DEBUG
+      --providers.docker=true
+      --providers.docker.exposedbydefault=false
+      --providers.file.filename=/dynamic.yaml
+      --providers.docker.network=traefik_proxy
+      --entrypoints.traefik_proxy.address=:80
+      --entrypoints.traefik_proxy-secured.address=:443
+      --certificatesresolvers.mytlschallenge.acme.tlschallenge=true
+      --certificatesresolvers.mytlschallenge.acme.email=admin@innoactive.de
+      --certificatesresolvers.mytlschallenge.acme.storage=/letsencrypt/acme.json
     volumes:
-      - /etc/nginx/conf.d
-      - /etc/nginx/vhost.d
-      - /usr/share/nginx/html
-      - /docker-gen-templates
-      - "{{ volume_names.tls_certificates }}:/etc/nginx/certs"
+      - ./letsencrypt:/letsencrypt # <== Volume for certs (TLS)
+      - /var/run/docker.sock:/var/run/docker.sock # <== Volume for docker admin
+      - ./dynamic.yaml:/dynamic.yaml # <== Volume for dynamic conf file, **ref: line 27
+
+    labels: {
+      traefik.enable: "true",
+      traefik.http.routers.https.rule: "HostRegexp(`{subdomain:.+}{{ hostname.split(',') | join('`, `{subdomain:.+}') }}`)
+      || Host(`{{ hostname.split(',') | join('`, `') }}`)",
+      traefik.http.routers.https.entryPoints: "traefik_proxy",
+      traefik.http.middlewares.https-redirect.redirectscheme.scheme: "https",
+      traefik.http.middlewares.https-redirect.redirectscheme.permanent: "true",
+      traefik.http.routers.https.middlewares: "https-redirect",
```

Comment:
we should probably set the redirect middleware per container, not globally (define globally, assign locally), so the global router is not required